cub/test/catch2_test_block_scan_api.cu

```diff
@@ -0,0 +1,143 @@
+ *
+ ******************************************************************************/
+
+#include <cub/block/block_scan.cuh>
+
+#include <thrust/detail/raw_pointer_cast.h>
+#include <thrust/device_vector.h>
+#include <thrust/host_vector.h>
+
+#include <cuda/std/numeric>
+
+#include "catch2_test_helper.h"
+
+constexpr int num_items_per_thread = 2;
+constexpr int block_num_threads    = 64;
+
+// example-begin inclusive-scan-array-init-value
+__global__ void InclusiveScanKernel(int* output)
+{
+  // Specialize BlockScan for a 1D block of 128 threads of type int
+  using block_scan_t   = cub::BlockScan<int, 128>;
+  using temp_storage_t = block_scan_t::TempStorage;
+
+  // Allocate shared memory for BlockScan
+  __shared__ temp_storage_t temp_storage;
+
+  int initial_value = 1;
+  int thread_data[] = {
+    +1 * ((int) threadIdx.x * num_items_per_thread), // item 0
+    -1 * ((int) threadIdx.x * num_items_per_thread + 1) // item 1
+  };
+  //  input: {[0, -1], [2, -3],[4, -5], ... [126, -127]}
+
+  // Collectively compute the block-wide inclusive prefix max scan
+  block_scan_t(temp_storage).InclusiveScan(thread_data, thread_data, initial_value, cub::Max());
+
+  // output: {[1, 1], [2, 2],[3, 3], ... [126, 126]}
+  // ...
+  // example-end inclusive-scan-array-init-value
+  output[threadIdx.x * 2]     = thread_data[0];
+  output[threadIdx.x * 2 + 1] = thread_data[1];
+}
+
+CUB_TEST("Block array-based inclusive scan works with initial value", "[block]")
+{
+  thrust::device_vector<int> d_out(block_num_threads * num_items_per_thread);
+
+  InclusiveScanKernel<<<1, block_num_threads>>>(thrust::raw_pointer_cast(d_out.data()));
+  REQUIRE(cudaSuccess == cudaPeekAtLastError());
+  REQUIRE(cudaSuccess == cudaDeviceSynchronize());
+
+  thrust::host_vector<int> expected(d_out.size());
+  for (size_t i = 0; i < expected.size() - 1; i += 2)
+  {
+    expected[i]     = static_cast<int>(i);
+    expected[i + 1] = static_cast<int>(i);
+  }
+
+  // When initial value = 1 for the given input the first two
+  // elements of the result are equal to 1.
+  expected[0] = 1;
+  expected[1] = 1;
+
+  REQUIRE(expected == d_out);
+}
+
+// example-begin inclusive-scan-array-aggregate-init-value
+__global__ void InclusiveScanKernelAggregate(int* output, int* block_aggregate)
+{
+  // Specialize BlockScan for a 1D block of 128 threads of type int
+  using block_scan_t   = cub::BlockScan<int, 128>;
+  using temp_storage_t = block_scan_t::TempStorage;
+
+  // Allocate shared memory for BlockScan
+  __shared__ temp_storage_t temp_storage;
+
+  int initial_value = 1;
+  int thread_data[] = {
+    +1 * ((int) threadIdx.x * num_items_per_thread), // item 0
+    -1 * ((int) threadIdx.x * num_items_per_thread + 1) // item 1
+  };
+  //  input: {[0, -1], [2, -3],[4, -5], ... [126, -127]}
+
+  // Collectively compute the block-wide inclusive prefix max scan
+  block_scan_t(temp_storage).InclusiveScan(thread_data, thread_data, initial_value, cub::Max(), *block_aggregate);
```

Comment:
question: I think `*block_aggregate` might be a bit confusing in the docs. Ideally, user should be able to copy-paste the code snippet and get working code. Do you think the code below can be any better from this perspective?
```suggestion
  int block_aggregate;
  block_scan_t(temp_storage).InclusiveScan(thread_data, thread_data, initial_value, cub::Max(), block_aggregate);
```